src/reference/asciidoc/scatter-gather.adoc

```diff
@@ -206,3 +206,8 @@ public Message<?> processAsyncScatterError(MessagingException payload) {
 To produce a proper reply, we have to copy headers (including `replyChannel` and `errorChannel`) from the `failedMessage` of the `MessagingException` that has been sent to the `scatterGatherErrorChannel` by the `MessagePublishingErrorHandler`.
 This way the target exception is returned to the gatherer of the `ScatterGatherHandler` for reply messages group completion.
 Such an exception `payload` can be filtered out in the `MessageGroupProcessor` of the gatherer or processed other way downstream, after the scatter-gather endpoint.
+
+NOTE: Before sending scattering results to the gatherer, `ScatterGatherHandler` reinstates the request message headers, including reply and error channels if any.
+This way errors from the `AggregatingMessageHandler` are going to be propagated to the caller, even if async hand off is applied for scatter recipients.
```

Comment:
`s/if async/if an async/`

`s/applied for scatter recipients/applied in scatter recipient subflows/` ?